packages/next/src/server/image-optimizer.ts

```diff
@@ -566,40 +593,73 @@ export async function imageOptimizer(
       'dangerouslyAllowSVG' | 'minimumCacheTTL'
     >
   },
-  isDev: boolean | undefined
-): Promise<{ buffer: Buffer; contentType: string; maxAge: number }> {
+  isDev: boolean | undefined,
+  previousCacheEntry?: IncrementalCacheItem
+): Promise<{
+  buffer: Buffer
+  contentType: string
+  maxAge: number
+  etag: string
+  upstreamEtag: string
+}> {
   const { href, quality, width, mimeType } = paramsResult
-  const upstreamBuffer = imageUpstream.buffer
+  const { buffer: upstreamBuffer } = imageUpstream
   const maxAge = getMaxAge(imageUpstream.cacheControl)
+
+  if (shouldUsePreviouslyCachedEntry(imageUpstream, previousCacheEntry)) {
+    // Return the cached optimized image
+    return {
+      buffer: previousCacheEntry.value.buffer,
+      contentType: paramsResult.mimeType,
+      maxAge: previousCacheEntry.curRevalidate || maxAge,
+      etag: previousCacheEntry.value.etag,
+      upstreamEtag: previousCacheEntry.value.upstreamEtag,
+    }
+  }
+
   const upstreamType =
     detectContentType(upstreamBuffer) ||
     imageUpstream.contentType?.toLowerCase().trim()
 
-  if (upstreamType) {
-    if (
-      upstreamType.startsWith('image/svg') &&
-      !nextConfig.images.dangerouslyAllowSVG
-    ) {
-      Log.error(
-        `The requested resource "${href}" has type "${upstreamType}" but dangerouslyAllowSVG is disabled`
-      )
-      throw new ImageError(
-        400,
-        '"url" parameter is valid but image type is not allowed'
-      )
-    }
+  if (
```

Comment:
Why was this svg logic moved?

Reply:
Seems like this a leftover, or don't remember anymore Why I moved it, seems like there's no point in doing that. Reverted this now